src/parser.rs

```diff
@@ -299,17 +299,31 @@ impl<'tokens, 'src> Parser<'tokens, 'src> {
     let mut items = Vec::new();
 
     let mut doc = None;
+    let mut comments: Vec<&'src str> = vec![];
```

Comment:
Is the lifetime annotation necessary? Could this just be:

```
let mut comments: Vec<&str> = vec![];
```

Also, when creating empty vecs, I usually use `Vec::new()`:

```suggestion
    let mut comments: Vec<&str> = Vec::new();
```